src/InstrumentationEngine/AppDomainCollection.h

```diff
@@ -55,7 +55,7 @@ namespace MicrosoftInstrumentationEngine
         // NOTE: These helpers will search all appdomains
         STDMETHOD(GetAssemblyInfoById)(_In_ AssemblyID assemblyID, _Out_ IAssemblyInfo** ppAssemblyInfo) override;
         STDMETHOD(GetModuleInfoById)(_In_ ModuleID moduleID, _Out_ IModuleInfo** ppModuleInfo) override;
-        STDMETHOD(GetModuleInfosByMvid)(_In_ GUID mvid, _Out_ IEnumModuleInfo** ppEnum) override;
+        STDMETHOD(GetModuleInfosByMvid)(GUID mvid, _Out_opt_ IEnumModuleInfo** ppEnum) override;
```

Comment:
>GUID mvid [](start = 40, length = 9)

Should this be, _In_ GUID& instead? or _In_ REFGUID?

Reply:
Ugh!!! but that will be a change to the interface. That was a bad mistake in the interface definition :-(. Do we need to udate the idl to fix this?

---
In reply to: [520723341](https://github.com/microsoft/CLRInstrumentationEngine/pull/325#discussion_r520723341) [](ancestors = 520723341)

Reply:
Yes, I guess we won't be able to change it without breaking compatibility .

Reply:
We don't want to break compatibility, for sure. Oh well, this is unfortunate.

---
In reply to: [521739359](https://github.com/microsoft/CLRInstrumentationEngine/pull/325#discussion_r521739359) [](ancestors = 521739359)